CONTRIBUTING.md

```diff
@@ -172,21 +175,9 @@ make integration-test
 Note that integration tests can take a long time to complete. Also the repositories that we clone for these tests will be stored under `output/` directory. If you do not remove/move this directory and run the pre-commit tool you might get errors.
 
 
-#### Generating documentation
-
-As mentioned above, all package code should make use of [Python docstrings](https://www.python.org/dev/peps/pep-0257/) in [reStructured text format](https://www.python.org/dev/peps/pep-0287/). Using these docstrings and the documentation template in the `docs/source/` folder, you can then generate proper documentation in different formats using the [Sphinx](https://github.com/sphinx-doc/sphinx/) tool:
-
-```bash
-make docs
-```
-
-This example generates documentation in HTML, which can then be found here:
-
-```bash
-open docs/_build/html/index.html
-```
```

Comment:
I initially wanted to remove this section to centralize instructions for generating documentation in one place: `docs/README.md`, and someone who needs to generate documentation will need to read this whole file.

If we want to keep this section, then we need to fix it a little bit. The first paragraph mentions the fact that we generate API references from docstrings. However, the following is the `make docs` command, which is not related to the API references pages. To generate these API reference pages, `sphinx-apidoc` or `make api-docs` needs to be run before `make 
docs`.

Another small detail is here, we only mention the fact that docstrings should follow the reStructuredText format, without mentioning the `numpydoc` style. I think the fact that we follow `numpydoc` style should also be mentioned.

Let me know what you think. If you agree, then I will proceed to modify this section with the changes I mentioned.

Reply:
> The first paragraph mentions the fact that we generate API references from docstrings. However, the following is the `make docs` command, which is not related to the API references pages. To generate these API reference pages, `sphinx-apidoc` or `make api-docs` needs to be run before `make docs`.

I don't think that's correct. `make docs` does generate html files using the docstrings it finds from the source code. If developers add a new module or remove it they need to call `make api-docs` first, otherwise `make docs` is enough.

Reply:
> Another small detail is here, we only mention the fact that docstrings should follow the reStructuredText format, without mentioning the `numpydoc` style. I think the fact that we follow `numpydoc` style should also be mentioned.

Good point. Please mention `numpydoc` style here too.

Reply:
> `make docs` does generate html files using the docstrings it finds from the source code. If developers add a new module or remove it they need to call `make api-docs` first to adjust the skeleton files, otherwise `make docs` is enough.
Ah, yes that is correct.

Reply:
See 4ce18a986997ad1fe4e3e6348f820a927d396e33.